### PR TITLE
Introduce and run pre-commit

### PR DESCRIPTION
**Pull request by Mojken**

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+      - id: end-of-file-fixer
+      - id: trailing-whitespace
+      - id: debug-statements
+      - id: detect-private-key
+  - repo: https://github.com/asottile/pyupgrade
+    rev: v2.31.0
+    hooks:
+      - id: pyupgrade
+        args:
+          - --py37-plus
```
Comment (Swamii) [inline .pre-commit-config.yaml:18]:
```suggestion
          - --py36-plus
```

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,81 @@
+default_language_version:
+  python: python3.10
```
Comment (Swamii) [inline .pre-commit-config.yaml:2]:
```suggestion
  python: python3.9
```

**Review by Swamii — COMMENTED** [review]
Great 👍

**Comment by Mojken** [comment]
This is what flake8 complained about btw:
```
runtests.py:15:20: C408 Unnecessary dict call - rewrite as a literal.
DEFAULT_SETTINGS = dict(
                   ^
djedi/__init__.py:36:14: C408 Unnecessary dict call - rewrite as a literal.
    config = dict(
             ^
djedi/admin/__init__.py:20:17: C408 Unnecessary dict call - rewrite as a literal.
                dict(
                ^
docs/conf.py:185:5: E265 block comment should start with '# '
    #'papersize': 'letterpaper',
    ^
docs/conf.py:187:5: E265 block comment should start with '# '
    #'pointsize': '10pt',
    ^
docs/conf.py:189:5: E265 block comment should start with '# '
    #'preamble': '',
    ^
djedi/compat.py:99:1: E402 module level import not at top of file
from collections import namedtuple
^
djedi/compat.py:100:1: E402 module level import not at top of file
from inspect import getfullargspec
^
```

**Comment by Mojken** [comment]
Closing in favor of #93

**Pull request closed without merging**